nms/app/packages/magmalte/app/views/equipment/FEGGatewayDetailSubscribers.js

```diff
@@ -0,0 +1,138 @@
+import type {subscriber} from '@fbcnms/magma-api';
+
+import ActionTable from '../../components/ActionTable';
+import FEGSubscriberContext from '../../components/context/FEGSubscriberContext';
+import Link from '@material-ui/core/Link';
+import LoadingFiller from '@fbcnms/ui/components/LoadingFiller';
+import React from 'react';
+import nullthrows from '@fbcnms/util/nullthrows';
+
+import {FetchSubscribers} from '../../state/lte/SubscriberState';
+import {
+  REFRESH_INTERVAL,
+  RefreshTypeEnum,
+  useRefreshingContext,
+} from '../../components/context/RefreshContext';
+import {useEffect, useState} from 'react';
+import {useRouter} from '@fbcnms/ui/hooks';
+
+type SubscriberRowType = {
+  name: string,
+  id: string,
+  service: string,
+};
+
+/**
+ * Returns a table of subscribers serviced by the federation gateway.
+ *
+ * @param {FEGGatewayDetailType} props An object containing the federation gateway and a boolean
+ *  telling to refresh or not
+ */
+export default function GatewayDetailSubscribers(props: FEGGatewayDetailType) {
+  const {history, match} = useRouter();
+  const networkId: string = nullthrows(match.params.networkId);
+  const [subscriberRows, setSubscriberRows] = useState<
+    Array<SubscriberRowType>,
+  >([]);
+  const [isLoading, setIsLoading] = useState(true);
+  // Auto refresh every REFRESH_INTERVAL seconds
+  const ctx = useRefreshingContext({
+    context: FEGSubscriberContext,
+    networkId: networkId,
+    type: RefreshTypeEnum.FEG_SUBSCRIBER,
+    interval: REFRESH_INTERVAL,
+    refresh: props.refresh,
+  });
+  const sessionState = ctx?.sessionState || {};
+  const subscriberToNetworkIdMap = {};
+
+  Object.keys(sessionState).map(servicedNetworkId => {
+    // $FlowIgnore[prop-missing] because refresh context returns other things too like state, enbInfo and each have their own property
+    const servicedNetworkSessionState = sessionState[servicedNetworkId] ?? {};
+    Object?.keys(servicedNetworkSessionState).map(subscriberImsi => {
+      subscriberToNetworkIdMap[subscriberImsi] = servicedNetworkId;
+    });
+  });
+  // get all the subscribers IMSI number serviced by the federation network
+  const subscribersImsi = JSON.stringify(Object.keys(subscriberToNetworkIdMap));
+
+  useEffect(() => {
+    const fetchSubscribersInfo = async () => {
+      const newSubscriberRows = [];
+      await Promise.all(
+        Object.keys(subscriberToNetworkIdMap).map(async subscriberImsi => {
+          // $FlowIgnore[incompatible-call] because it can be called with different values when getting paginated subscribers
+          const subscriberInfo: subscriber = await FetchSubscribers({
+            networkId: subscriberToNetworkIdMap[subscriberImsi],
+            id: subscriberImsi,
+          });
+          newSubscriberRows.push({
+            name: subscriberInfo?.name || subscriberImsi,
+            id: subscriberImsi,
+            service: subscriberInfo?.lte?.state || '-',
+          });
+        }),
+      );
```

Comment:
Could you add a TODO referencing my name `andreilee` saying to update this to use a bulk fetch API?